src/mbed_tools/project/project.py

```diff
@@ -43,20 +43,20 @@ def initialise_project(path: pathlib.Path, create_only: bool) -> None:
         program.resolve_libraries()
 
 
-def checkout_project_revision(path: pathlib.Path, force: bool = False) -> None:
-    """Checkout a specific revision of the current Mbed project.
+def deploy_project_revision(path: pathlib.Path, force: bool = False) -> None:
```

Comment:
`deploy_project_revision` doesn't make sense. We should just leave the API names how they are, or at least come up with something that makes sense.

Reply:
Given that this is the public API we could change the name to `deploy` here. We should definitely change it from `deploy_project_revision` as that doesn't make sense to a reader of this code.

Reply:
I think it's the same as 'deploy a project revision' or 'checkout a project revision'. Naming it to `deploy` would create confusion as there is already a definition/function with the same.

Reply:
With your changes we already have multiple functions and methods named deploy. They are all namespaced in their own modules though.

The function names before and after your changes:
```
cli.checkout -> cli.deploy
project.checkout_project_revision -> project.deploy_project_revision
project.MbedProgram.checkout_libraries -> project.MbedProgram.deploy_libraries
project._internal.LibraryReferences.checkout -> project._internal.LibraryReferences.deploy
```
So you can see you already have multiple functions named deploy now (project._internal.LibraryReferences.deploy & cli.deploy). So, I'm not sure I'm getting your point about "creating confusion". All I know is the name `deploy_project_revision` is already creating confusion for me as it doesn't make much sense as a phrase.